bolt-sidecar/bin/sidecar.rs

```diff
@@ -1,140 +1,167 @@
-    let consensus_state = ConsensusState::new(&config.beacon_api_url);
+    let shutdown_tx = start_rpc_server(&config, api_events).await?;
+
+    let mut consensus_state =
+        ConsensusState::new(&config.beacon_api_url, config.commitment_deadline);
+
+    // TODO: this can be replaced with ethereum_consensus::clock::from_system_time()
+    // but using beacon node events is easier to work on a custom devnet for now
+    // (as we don't need to specify genesis time and slot duration)
+    let mut head_tracker = HeadTracker::start(&config.beacon_api_url);
 
     let builder_proxy_config = BuilderProxyConfig {
-        mevboost_url: config.mevboost_url,
+        mevboost_url: config.mevboost_url.clone(),
         server_port: config.mevboost_proxy_port,
     };
 
     let (payload_tx, mut payload_rx) = mpsc::channel(16);
     let payload_fetcher = LocalPayloadFetcher::new(payload_tx);
 
+    let mut local_builder = LocalBuilder::new(&config);
+
     tokio::spawn(async move {
-        loop {
-            if let Err(e) =
-                start_builder_proxy(payload_fetcher.clone(), builder_proxy_config.clone()).await
-            {
-                tracing::error!("Builder API proxy failed: {:?}", e);
-                tokio::time::sleep(Duration::from_secs(5)).await;
-            }
+        if let Err(e) = start_builder_proxy_server(payload_fetcher, builder_proxy_config).await {
+            tracing::error!("Builder API proxy failed: {:?}", e);
         }
     });
 
     // TODO: parallelize this
     loop {
         tokio::select! {
-            Some(event) = api_events_rx.recv() => {
-                tracing::info!("Received commitment request: {:?}", event.request);
-                let request = event.request;
+            Some(ApiEvent { request, response_tx }) = api_events_rx.recv() => {
+                tracing::info!("Received commitment request: {:?}", request);
 
-                if let Err (e) = consensus_state.validate_request(&CommitmentRequest::Inclusion(request.clone())) {
-                    tracing::error!("Failed to validate request: {:?}", e);
-                    let _ = event.response.send(Err(ApiError::Custom(e.to_string())));
+                if let Err (e) = consensus_state.validate_request(&request) {
+                    tracing::error!(err = ?e, "Consensus: failed to validate request");
+                    let _ = response_tx.send(Err(ApiError::Custom(e.to_string())));
                     continue;
                 }
 
-                if let Err(e) = execution_state
-                    .try_commit(&CommitmentRequest::Inclusion(request.clone()))
-                    .await
-                {
-                    tracing::error!("Failed to commit request: {:?}", e);
-                    let _ = event.response.send(Err(ApiError::Custom(e.to_string())));
-                    continue;
-                }
+                // if let Err(e) = execution_state
```

Comment:
Note: let's see if we can enable this without too much trouble!

Reply:
done in latest commit by @thedevbirb